Sources/MySQLKit/MySQLDataDecoder.swift

```diff
@@ -1,36 +1,36 @@
 import Foundation
 import MySQLNIO
 import NIOFoundationCompat
+@_spi(CodableUtilities) import SQLKit
 
-extension Optional: MySQLDataConvertible where Wrapped: MySQLDataConvertible {
-    public init?(mysqlData: MySQLData) {
-        if mysqlData.buffer != nil {
-            guard let value = Wrapped.init(mysqlData: mysqlData) else {
-                return nil
-            }
-            self = .some(value)
-        } else {
-            self = .none
-        }
-    }
+/// Translates `MySQLData` values received from the database into `Decodable` values.
+///
+/// Types which conform to `MySQLDataConvertible` are converted directly to the requested type. For other types,
+/// an attempt is made to interpret the database value as JSON and decode the type from it.
+public struct MySQLDataDecoder: Sendable {
+    /// A wrapper to silence `Sendable` warnings for `JSONDecoder` when not on macOS.
+    struct FakeSendable<T>: @unchecked Sendable { let value: T }
     
-    public var mysqlData: MySQLData? {
-        return self.flatMap(\.mysqlData)
-    }
-}
-
-public struct MySQLDataDecoder {
-    let json: JSONDecoder
-
+    /// The `JSONDecoder` used for decoding values that can't be directly converted.
+    let json: FakeSendable<JSONDecoder>
```

Comment:
We could just `@preconcurrency import Foundation` on Linux instead

Reply:
That triggers a whole different mess of warnings, since Sendability on macOS isn't available before macOS 13.0. This is actually much easier.